### PR TITLE
[DTP-1078] Use lexico timeserials and `siteCode` field in StateMessages

### PR DESCRIPTION
**Pull request by VeskeR**
Resolves [DTP-1078](https://ably.atlassian.net/browse/DTP-1078)

[DTP-1078]: https://ably.atlassian.net/browse/DTP-1078?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Release Notes

- **New Features**
	- Introduced a new optional `siteCode` property in the `StateMessage` class for enhanced message tracking.
	- Added a utility function `lexicoTimeserial` for consistent timestamp formatting in state operation messages.

- **Improvements**
	- Streamlined handling of timeserials across various classes, improving clarity and efficiency.
	- Updated logging to reference the new site code in operations.
	- Enhanced documentation for the `timeserial` property in the `StateMapEntry` interface.

- **Bug Fixes**
	- Enhanced validation and error handling in the `LiveObject` class for timeserial comparisons.

- **Tests**
	- Refactored test cases to utilize the new `lexicoTimeserial` function and include `siteCode` in state operation messages, improving test clarity and maintainability.

- **Chores**
	- Removed the `timeserial.ts` file from the allowed files for the LiveObjects plugin, refining bundle size management.

### CODE REVIEW

File: src/plugins/liveobjects/livemap.ts [hunk 1]
```diff
@@ -455,6 +450,37 @@ export class LiveMap<T extends API.LiveMapType> extends LiveObject<LiveMapData,
     return { update: { [op.key]: 'removed' } };
   }
 
+  /**
+   * Returns true if the origin timeserials of the given operation and entry indicate that
+   * the operation should be applied to the entry, following the CRDT semantics of this LiveMap.
+   */
+  private _canApplyOperationFollowingCRDTRules(
```
Comment (mschristensen) [inline src/plugins/liveobjects/livemap.ts:457]: Would a better name for this be `_canApplyMapOperation`?

**Review by mschristensen — APPROVED** [review]
Looks great

File: src/plugins/liveobjects/livemap.ts (hunk 1 above)
Comment (VeskeR) [reply src/plugins/liveobjects/livemap.ts:457]: yes! I was struggling with naming this one 😅 , and just saying `Map` is better for sure, thanks!

Comment (VeskeR) [reply src/plugins/liveobjects/livemap.ts]: updated

File: test/realtime/live_objects.test.js [hunk 2]
```diff
@@ -2043,49 +2075,46 @@ define(['ably', 'shared_helper', 'chai', 'live_objects', 'live_objects_helper'],
...
+            liveObjectsHelper,
+            channelName,
+            channel,
+            sampleMapKey,
+            sampleMapObjectId,
+            sampleCounterKey,
+            sampleCounterObjectId,
+          });
+        }, client);
+      });
```
[the remark this answers is not in the document]
Comment (VeskeR) [reply test/realtime/live_objects.test.js:2117]: all tests inherently have a 60 seconds timeout from the mocha, no need to add it manually everywhere

**Pull request merged**